tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+
+So, now we know what golden testing is, but if unit tests and golden tests are so similar, why don't we just keep up with unit tests? Right?
+
+
+The truth is that unit tests are not useful when evaluating large, complex outputs and that's where the difference lies. Tests are all about maintainability and readability, and when we test large outputs we tend to add visual noise to our code. Golden Tests helpfully fix this issue. Let's check the main advantages:
+
+ - Large outputs are stored in a golden file reducing the visual noise in our code.
+
+ - With some golden tests libraries the Golden Files can be automatically generated.
+
+ - Some golden test libraries also automatically update the golden file. So we don't need to manually update the tests when the expected output changes. 
+
+ - It is useful when working with JSON, HTML, images, etc.
+
+In this tutorial we will use [hspec-golden](http://hackage.haskell.org/package/hspec-golden) library to build golden tests. We will also use the `hspec-golden` CLI to easily update them.
+
+You can either code along with this tutorial or check out the finished [code](https://github.com/stackbuilders/tutorials/tree/tutorials/tutorials/haskell/hspec-golden-test-tutorial/code) in Github. 
+
+## Hspec-golden
+
+`Hspec-golden` is a StackBuilders testing library written in Haskell, that helps users implement golden tests. It is an open source project so feel free to [review](https://github.com/stackbuilders/hspec-golden) it.
+
+## Getting Started
+
+First things first, let's create a new Haskell project to work in:
+
+```bash
+stack new "hspec-golden-tests-tutorial"
+```
+
+and add the necessary dependencies in order to follow this tutorial without issues.
+
+Project dependencies:
+
+```yaml
+dependencies:
+- base >= 4.7 && < 5
+- aeson
+- text
+- bytestring
+- blaze-html
+```
+
+Testing dependencies:
+
+```yaml
+dependencies:
+- hspec-golden-tests
+- hspec
+- hspec-golden
+
+```
+## A Simple Test
+
+Now that we have all of the dependencies installed, let's get started with the golden tests.
+
+First let's create a `FizzBuzz ` module to be tested and add some code. For convenience, I have created it inside a `FIZZBUZZ` directory. 
+
+```haskell
+module FIZZBUZZ.FizzBuzz where 
+
+fizzBuzz :: [Int] -> [String]
+fizzBuzz list = map fizzOrBuzz list
+
+fizzOrBuzz ::Int -> String
+fizzOrBuzz n | n `mod` 15 == 0  = "FizzBuzz"
+             | n `mod` 3  == 0  = "Fizz"
+             | n `mod` 5  == 0  = "Buzz"
+             | otherwise        = show n
+
+```
+For simplicity we will test a simple function `fizzBuzz` that replaces multiples of 3 with Fizz and multiples of 5 with Buzz. In case that a number is a multiple of both it replaces the number with FizzBuzz. Before moving on with testing, let's review some of `hspec-golden` [documentation](http://hackage.haskell.org/package/hspec-golden-0.1.0.1/docs/Test-Hspec-Golden.html).
```

Comment:
```suggestion
For simplicity, we will test a simple function `fizzBuzz` that replaces the multiples of 3 with "Fizz" and the multiples of 5 with "Buzz". In case that a number is a multiple of both, it replaces the number with "FizzBuzz". Before we continue with testing, let's review some of `hspec-golden` [documentation](http://hackage.haskell.org/package/hspec-golden-0.1.0.1/docs/Test-Hspec-Golden.html).
```